nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/ConsumeMQTT.java

```diff
@@ -383,9 +383,8 @@ private void initializeClient(ProcessContext context) {
         // non-null but not connected, so we need to handle each case and only create a new client when it is null
         try {
             mqttClient = createMqttClient();
-            mqttClient.setCallback(this);
             mqttClient.connect();
-            mqttClient.subscribe(topicPrefix + topicFilter, qos);
+            mqttClient.subscribe(topicPrefix + topicFilter, qos, this);
```

Comment:
It's uneasy to follow the code here. A processor class being a callback itself is strange. What do you think of implementing the interface as an anonymous class instead of passing 'this'?

Reply:
It was a consequence of the original implementation, but you are right. We no longer need to implement the interface on the processor.